decidim-core/app/helpers/decidim/short_link_helper.rb

```diff
@@ -22,7 +22,7 @@ def short_url(**kwargs)
       target ||= Rails.application
 
       mounted_engine = EngineResolver.new(_routes).mounted_name
-      ShortLink.to(target, mounted_engine, **kwargs).short_url
+      ShortLink.to(target, target.try(:mounted_engine) || mounted_engine, **kwargs).short_url
```

Comment:
I would change this so you can avoid the call to the EngineResolver if you currently have a `mounted_engine` in the `target`:

```suggestion
      mounted_engine = target.try(:mounted_engine) || EngineResolver.new(_routes).mounted_name
      ShortLink.to(target, mounted_engine, **kwargs).short_url
```

Reply:
fixed in [4589d1f](https://github.com/decidim/decidim/pull/12004/commits/4589d1f77f8acc98cf02fd56b07adee79fc4a665)